### PR TITLE
drivers: clock_control: Move stm32f1 series to Cube LL API

### PR DESCRIPTION
**Pull request by erwango**
stm32f1 series migrated to Cube LL API.
This completes migration of stm32 clock control drivers to unique API and hence simplify generic
driver implementation in calls to clock control driver.
Former driver code cleaned up.

### CODE REVIEW

File: drivers/clock_control/Kconfig.stm32 [hunk 1]
```diff
@@ -222,11 +273,11 @@ config CLOCK_STM32_APB1_PRESCALER
...
 config CLOCK_STM32_APB2_PRESCALER
 	int "APB2 prescaler"
-	depends on CLOCK_CONTROL_STM32_CUBE
 	default 1
 	range 1 16
 	help
 	  APB2 High speed clock (PCLK2) prescaler, allowed values:
 	  1, 2, 4, 8, 16
 
+endif #
```
Comment (galak) [inline drivers/clock_control/Kconfig.stm32]: end if #CLOCK_CONTROL_STM32_CUBE

**Review by galak — CHANGES_REQUESTED** [review]
minor change to Kconfig comment

File: arch/arm/soc/st_stm32/stm32f1/Kconfig.defconfig.series [hunk 2]
```diff
@@ -12,6 +12,13 @@ source "arch/arm/soc/st_stm32/stm32f1/Kconfig.defconfig.stm32f1*"
 config SOC_SERIES
 	default stm32f1
 
+if CLOCK_CONTROL
```
Comment (galak) [inline arch/arm/soc/st_stm32/stm32f1/Kconfig.defconfig.series]: Can we do:

select CLOCK_CONTROL_STM32_CUBE if CLOCK_CONTROL

in arch/arm/soc/st_stm32/Kconfig so we don't need to duplicate this everywhere

**Review by galak — CHANGES_REQUESTED** [review]
Some Kconfig possibly cleanup

**Review by galak — CHANGES_REQUESTED** [review]
Should gpio before remove of the native ll change?

File: arch/arm/soc/st_stm32/stm32f1/Kconfig.defconfig.series (hunk 2 above)
Comment (erwango) [inline arch/arm/soc/st_stm32/stm32f1/Kconfig.defconfig.series]: ok, will do

File: drivers/clock_control/Kconfig.stm32 (hunk 1 above)
Comment (erwango) [inline drivers/clock_control/Kconfig.stm32]: Done. Thanks.

**Comment by erwango** [comment]
About GPIO:
It is needed to get gpio_stm32.h modif available when activating CONFIG_CLOCK_CONTROL_STM32_CUBE in in boards defconfig.

The gpio_stm32.c could be moved in clean up, but I decided to group .h and .c gpio stuff

**Comment by galak** [comment]
For the gpio stuff, I just meant move the commit earlier in the series of commits before the board changes to enable the CUBE clock support

**Review by galak — APPROVED** [review]

**Pull request merged**